sdk/textanalytics/Azure.AI.TextAnalytics/tests/TextAnalyticsRecordedTestSanitizer.cs

```diff
@@ -0,0 +1,39 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Azure.Core.Testing
+{
+    public class TextAnalyticsRecordedTestSanitizer : RecordedTestSanitizer
+    {
+        public override string SanitizeUri(string uri)
+        {
+            var index = uri.IndexOf(".cognitiveservices.azure.com", StringComparison.Ordinal);
+            return index >= 0 ? "https://test" + uri.Substring(index) : base.SanitizeUri(uri);
+        }
+
+        public override void SanitizeHeaders(IDictionary<string, string[]> headers)
+        {
+            const string key = "Ocp-Apim-Subscription-Key";
+            if (headers.ContainsKey(key))
+            {
+                headers[key] = new[] { SanitizeValue };
+            }
+
+            base.SanitizeHeaders(headers);
+        }
+
+        public override string SanitizeVariable(string variableName, string environmentVariableValue)
+        {
+            return variableName switch
+            {
+                "TEXT_ANALYTICS_ENDPOINT" => "https://test.cognitiveservices.azure.com/",
```

Comment:
Do everyone use the same endpoint when recording?

Reply:
AFAIK, it is only used in `Azure.AI.TextAnalytics`

Reply:
Not really, I've seen Alexander and Anne using a different one than mine.

Reply:
This kind of sanitization might not work then. If in playback more the client would always use `https://test.cognitiveservices.azure.com/` but recordings would have something else they wouldn't match.

Reply:
I think it should work in combination with `SanitizeUri`:
```cs
public override string SanitizeUri(string uri)
{
    var index = uri.IndexOf(".cognitiveservices.azure.com", StringComparison.Ordinal);
    return index >= 0 ? "https://test" + uri.Substring(index) : base.SanitizeUri(uri);
}
```

Reply:
But why do we need this complication?

Reply:
I prefer not to expose any private data if I can. If @annelo-msft is ok with her cognitive services URL prefix on github, I can remove this sanitization.

Reply:
My worry is that if the full uri is ever returned as part of the body (for example in paged responses) you won't be able to normalize it.

Reply:
>If @annelo-msft is ok with her cognitive services URL prefix on github

this will change according to who is running the records. If it is me, I use my own resource.
It might be someone external with their own resource too.
From what Pavel is saying, it looks like for now it might not be needed

Reply:
Correct me if I'm wrong, but doesn't the ARM work Heath was driving make it so we can always point to the same resource?  That way community members can run our tests without needed access to internal subscriptions?

Reply:
It is not related to the ARM template, it is about having the URL that was used in recorded session. But since we're not doing this type of sanitization anywhere else, I can remove it.

Reply:
Will my endpoint go in recordings then?

Reply:
Yes, that's how usually do it.